pkg/k8sutils/templates/pod/spec.go

```diff
@@ -0,0 +1,369 @@
+			Name: secret.Name,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secret.Name,
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      secret.Name,
+			MountPath: secret.MountPath,
+		})
+	}
+	return volumes, volumeMounts
+}
+
+func MountStorageVolumes(spec v1.SpecInterface) ([]corev1.Volume, []corev1.VolumeMount, map[string]bool) {
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	vexist := make(map[string]bool)
+	for _, sv := range spec.GetStorageVolumes() {
+		vexist[sv.MountPath] = true
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      sv.Name,
+			MountPath: sv.MountPath,
+		})
+
+		volumes = append(volumes, corev1.Volume{
+			Name: sv.Name,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: sv.Name,
+				},
+			},
+		})
+	}
+	return volumes, volumeMounts, vexist
+}
+
+func MountConfigMap(volumes []corev1.Volume, volumeMounts []corev1.VolumeMount,
+	cmInfo v1.ConfigMapInfo, mountPath string) ([]corev1.Volume, []corev1.VolumeMount) {
+	if cmInfo.ConfigMapName != "" && cmInfo.ResolveKey != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: cmInfo.ConfigMapName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cmInfo.ConfigMapName,
+					},
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      cmInfo.ConfigMapName,
+			MountPath: mountPath,
+		})
+	}
+	return volumes, volumeMounts
+}
+
+func Labels(clusterName string, spec v1.SpecInterface) map[string]string {
+	labels := statefulset.Selector(clusterName, spec)
+	switch v := spec.(type) {
+	case *v1.StarRocksBeSpec:
+		if v != nil {
+			labels.AddLabel(v.PodLabels)
+		}
+	case *v1.StarRocksCnSpec:
+		if v != nil {
+			labels.AddLabel(v.PodLabels)
+		}
+	case *v1.StarRocksFeSpec:
+		if v != nil {
+			labels.AddLabel(v.PodLabels)
+		}
+	}
+	return labels
+}
+
+func Envs(clusterName string, namespace string,
+	spec v1.SpecInterface, config map[string]interface{}, feExternalServiceName string) []corev1.EnvVar {
+	envs := []corev1.EnvVar{
+		{
+			Name: "POD_NAME",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"},
+			},
+		},
+		{
+			Name: "POD_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"},
+			},
+		},
+		{
+			Name: "HOST_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.hostIP"},
+			},
+		},
+		{
+			Name: "POD_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"},
+			},
+		},
+		{
+			Name:  "HOST_TYPE",
+			Value: "FQDN",
+		},
+		{
+			Name:  "USER",
+			Value: "root",
```

Comment:
// TODO: make this configurable.

Reply:
So, USER==root is its default value?

Reply:
yes, I mean add a TODO here, we will soon go back here to support non-root mode.